.github/workflows/codeql.yml

```diff
@@ -12,52 +12,62 @@
 name: "CodeQL"
 
 on:
-    push:
-        branches: '*'
-    pull_request:
-        # The branches below must be a subset of the branches above
-        branches: '*'
-    schedule:
-        -   cron: '44 9 * * 5'
+  push:
+  pull_request:
 
 jobs:
-    analyze:
-        name: "Analyze"
-        runs-on: "ubuntu-latest"
-        permissions:
-            actions: read
-            contents: read
-            security-events: write
-        strategy:
-            fail-fast: false
-            matrix:
-                language: [ 'java' ]
-                # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby' ]
-                # Use only 'java' to analyze code written in Java, Kotlin or both
-                # Use only 'javascript' to analyze code written in JavaScript, TypeScript or both
-                # Learn more about CodeQL language support at https://aka.ms/codeql-docs/language-support
-        steps:
-            -   uses: actions/checkout@v3.3.0
-            -   uses: gradle/wrapper-validation-action@v1.0.6
-            -   uses: actions/setup-java@v3.10.0
-                with:
-                    distribution: "zulu"
-                    java-version: "17"
-            # Initializes the CodeQL tools for scanning.
-            -   name: Initialize CodeQL
-                uses: github/codeql-action/init@v2
-                with:
-                    debug: true
-                    languages: ${{ matrix.language }}
-                    # If you wish to specify custom queries, you can do so here or in a config file.
-                    # By default, queries listed here will override any specified in a config file.
-                    # Prefix the list here with "+" to use these queries and those in the config file.
-                    # Details on CodeQL's query packs refer to : https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#using-queries-in-ql-packs
-                    # queries: security-extended,security-and-quality
-                    # Autobuild attempts to build any compiled languages  (C/C++, C#, Go, or Java).
-            -   name: Autobuild
-                uses: github/codeql-action/autobuild@v2
-            -   name: Perform CodeQL Analysis
-                uses: github/codeql-action/analyze@v2
-                with:
-                    category: "/language:${{matrix.language}}"
+  analyze:
+    name: "Analyze"
+    runs-on: "ubuntu-latest"
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'java' ]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Validate Gradle Wrapper
+        uses: gradle/actions/wrapper-validation@v3
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v4
+        with:
+          java-version: "8"
```

Comment:
Consider whether java version 8 is sufficient for building with gradle.

Reply:
It is since there's no Gradle plugin that requires it